app/src/androidTest/java/com/github/polypoly/app/base/game/PlayerTest.kt

```diff
@@ -0,0 +1,245 @@
+package com.github.polypoly.app.base.game
+
+import com.github.polypoly.app.base.game.bonus_card.BonusCard
+import com.github.polypoly.app.base.game.bonus_card.InGameBonusCard
+import com.github.polypoly.app.base.game.rules_and_lobby.GameLobby
+import com.github.polypoly.app.base.game.location.InGameLocation
+import com.github.polypoly.app.commons.PolyPolyTest
+import com.github.polypoly.app.commons.PolyPolyTest.Companion.TEST_USER_0
+import com.github.polypoly.app.map.LocationRepository
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.osmdroid.util.GeoPoint
+
+@RunWith(JUnit4::class)
+class PlayerTest {
+
+    private val gameLobby = GameLobby(PolyPolyTest.TEST_USER_1, PolyPolyTest.gameRulesDefault,
+        "test_game", "123456", false)
+
+    @Before
+    fun startGame() {
+        gameLobby.addUser(PolyPolyTest.TEST_USER_2)
+        gameLobby.addUser(PolyPolyTest.TEST_USER_3)
+        gameLobby.addUser(PolyPolyTest.TEST_USER_4)
+        gameLobby.addUser(PolyPolyTest.TEST_USER_5)
+        gameLobby.addUser(TEST_USER_0)
+        gameLobby.start()
+    }
+
+
+    @Test
+    fun greatestPlayerIsTheOneWithMoreMoney() {
+       assertTrue(PolyPolyTest.testPlayer1 < PolyPolyTest.testPlayer2)
+       assertTrue(PolyPolyTest.testPlayer2 > PolyPolyTest.testPlayer3)
+    }
+
+    @Test
+    fun playerWithSameBalanceGreaterThan0AreEquals() {
+        assertTrue(PolyPolyTest.testPlayer1.compareTo(PolyPolyTest.testPlayer3) == 0)
+    }
+
+    @Test
+    fun playerWhoLostFirstIsTheSmallest() {
+        assertTrue(PolyPolyTest.testPlayer4 < PolyPolyTest.testPlayer5)
+    }
+
+    @Test
+    fun compareToConsiderEqualThePlayerWhoLostAtTheSameTime() {
+        assertTrue(PolyPolyTest.testPlayer6.compareTo(PolyPolyTest.testPlayer5) == 0)
+    }
+
+    @Test
+    fun loseMoneyDecreaseTheMoneyOfThePlayer() {
+        val playerMoney = 100
+        val moneyLost = 50
+        val testPlayer = Player(TEST_USER_0, playerMoney, listOf())
+        testPlayer.loseMoney(moneyLost)
+        assertEquals(playerMoney - moneyLost, testPlayer.getBalance())
+    }
+
+    @Test
+    fun loseMoneyDoesNotDecreaseTheMoneyOfThePlayerBelowZero() {
+        val testPlayer = Player(TEST_USER_0, 100, listOf())
+        testPlayer.loseMoney(150)
+        assertEquals(0, testPlayer.getBalance())
+    }
+
+    @Test
+    fun loseMoneyThrowsAnExceptionIfTheAmountIsNegative() {
+        val testPlayer = Player(TEST_USER_0, 100, listOf())
+        val thrown = assertThrows(IllegalArgumentException::class.java) {
+            testPlayer.loseMoney(-50)
+        }
+        assertNotNull(thrown.message)
+        val message = thrown.message
+        if (message != null) {
```

Comment:
this has to be true given the assertNotNull earlier, right?

Reply:
Yes my bad, when I changed the tests I forgot to remove this part. Fix done